src/chains/ethereum/ethereum/src/database.ts

```diff
@@ -1,34 +1,51 @@
-import type { AbstractLevelDOWN } from "abstract-leveldown";
+import { AbstractSublevel } from "abstract-level";
 import Emittery from "emittery";
 import { dir, setGracefulCleanup } from "tmp-promise";
 import Blockchain from "./blockchain";
 import { EthereumInternalOptions } from "@ganache/ethereum-options";
-import sub from "subleveldown";
-import encode from "encoding-down";
-import leveldown from "leveldown";
-import type { LevelUp } from "levelup";
-const levelup = require("levelup");
+import { Level } from "level";
+import { UpgradedLevelDown } from "./leveldown-to-level";
+
+export type GanacheLevel = Level<Buffer, Buffer>;
+export type GanacheSublevel = AbstractSublevel<
+  GanacheLevel,
+  Buffer,
+  Buffer,
+  Buffer
+>;
 
 setGracefulCleanup();
 const tmpOptions = { prefix: "ganache_", unsafeCleanup: true };
 const noop = () => Promise.resolve();
 
+export enum DBType {
+  Level = 0,
+  LevelDown = 1
+}
+
+export const LEVEL_OPTIONS = {
+  keyEncoding: "binary" as const,
+  valueEncoding: "binary" as const,
+  // specify an empty `prefix` for browser-based leveldown (level-js)
+  prefix: "" as const
+};
 export default class Database extends Emittery {
   public readonly blockchain: Blockchain;
   readonly #options: EthereumInternalOptions["database"];
+  // TODO: implement cleanup https://github.com/trufflesuite/ganache/issues/3662
   #cleanupDirectory = noop;
   #closed = false;
   public directory: string = null;
-  public db: LevelUp = null;
-  public blocks: LevelUp;
-  public blockIndexes: LevelUp;
-  public blockLogs: LevelUp;
-  public transactions: LevelUp;
-  public transactionReceipts: LevelUp;
-  public storageKeys: LevelUp;
-  public trie: LevelUp;
+  public db: GanacheLevel;
+  public blocks: GanacheSublevel;
+  public blockIndexes: GanacheSublevel;
+  public blockLogs: GanacheSublevel;
+  public transactions: GanacheSublevel;
+  public transactionReceipts: GanacheSublevel;
+  public storageKeys: GanacheSublevel;
+  public trie: GanacheSublevel;
   public readonly initialized: boolean;
-  #rootStore: AbstractLevelDOWN;
+  public type: DBType = DBType.Level;
```

Comment:
I had that originally and am open to putting it back. I liked initializing to `DBType.Level` because 2 out of the 3 code paths in the `initialize` function end up being `Level`, only one is `LevelDown`. Eventually we'll support someone providing their _own_ level instance, so then it will be 3 out of 4 code paths that are all `Level`. So I think starting with `DBType.Level` gets across that this is our "default type". Open to disagreement if you still think it should change.